### PR TITLE
policy: simplify and consolidate some code

### PR DESCRIPTION
**Pull request by dcbw**
Simplify and consolidate policy code.

@squeed @girishmg @danwinship @alexanderConstantinescu @mccv1r0 @JacobTanenbaum

### CODE REVIEW

File: go-controller/pkg/ovn/common.go [hunk 1]
```diff
@@ -199,28 +198,6 @@ func (oc *Controller) deletePortGroup(hashName string) {
 	}
 }
 
-func (oc *Controller) getIPFromOvnAnnotation(ovnAnnotation string) string {
```
Comment (danwinship) [inline go-controller/pkg/ovn/common.go]: you could keep this function and just make it be a wrapper around UnmarshalPodAnnotation... (the fewer places I see `.IP.IP.String()` the better...)

File: go-controller/pkg/util/util.go [hunk 2]
```diff
@@ -45,65 +45,107 @@ func GetNodeChassisID() (string, error) {
...
 type PodRoute struct {
 	// Dest is the route destination
-	Dest string `json:"dest"`
+	Dest *net.IPNet
 	// NextHop is the IP address of the next hop for traffic destined for Dest
+	NextHop net.IP
+}
+
+// Internal struct used to correctly marshal IPs to JSON
+type _podAnnotation struct {
```
[the remark this answers is not in the document]
Comment (dcbw) [reply go-controller/pkg/util/util.go]: Fixed the _. I think I'd like to stick with the structs just so I don't have to handle Routes manipulation...

File: go-controller/pkg/ovn/common.go (hunk 1 above)
Comment (dcbw) [reply go-controller/pkg/ovn/common.go]: @danwinship can we keep it for now? It goes away in the upcoming PR I split this cleanup off from...

**Comment by dcbw** [comment]
@danwinship PTAL thanks!

File: go-controller/pkg/ovn/policy_common.go [hunk 3]
```diff
@@ -165,6 +168,201 @@ func (gp *gressPolicy) delAddressSet(hashedAddressSet string) (string, string, b
 	return oldL3Match, gp.getL3MatchFromAddressSet(), true
 }
 
+func (oc *Controller) handlePeerPodSelectorAddUpdate(np *namespacePolicy,
```
Comment (squeed) [inline go-controller/pkg/ovn/policy_common.go]: docblock please!

File: go-controller/pkg/ovn/policy.go [hunk 4]
```diff
@@ -672,56 +523,6 @@ func (oc *Controller) handlePeerNamespaceSelectorModify(
...
-		logrus.Errorf("error watching namespaces for policy %s: %v",
-			policy.Name, err)
-		return
-	}
-
-	np.nsHandlerList = append(np.nsHandlerList, h)
-
-}
-
 // AddNetworkPolicy adds network policy and create corresponding acl rules
```
Comment (squeed) [inline go-controller/pkg/ovn/policy.go]: this docblock is wrong

**Comment by danwinship** [comment]
lgtm

File: go-controller/pkg/ovn/policy_common.go [hunk 5]
```diff
@@ -165,6 +168,201 @@ func (gp *gressPolicy) delAddressSet(hashedAddressSet string) (string, string, b
...
+	addressMap map[string]bool,
+	np *namespacePolicy) {
+
+	namespaceHandler, err := oc.watchFactory.AddFilteredNamespaceHandler("",
+		namespaceSelector,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				namespace := obj.(*kapi.Namespace)
+				np.Lock()
+				alreadyDeleted := np.deleted
```
Comment (squeed) [inline go-controller/pkg/ovn/policy_common.go]: Theoretically, you should be comparing the UIDs to ensure this isn't a new namespace. We can let this slide for now (I know this is a bigger refactor)

File: go-controller/pkg/ovn/policy_common.go (hunk 3 above)
Comment (dcbw) [reply go-controller/pkg/ovn/policy_common.go]: @squeed it's copy & paste, there wasn't one before.

File: go-controller/pkg/ovn/policy_common.go (hunk 5 above)
Comment (dcbw) [reply go-controller/pkg/ovn/policy_common.go]: @squeed also no change from previous, it's just moving code.

File: go-controller/pkg/ovn/policy_common.go (hunk 3 above)
Comment (squeed) [reply go-controller/pkg/ovn/policy_common.go]: for sure, but at this point, you understand this function. if you could capture that, that'd be good.

**Comment by dcbw** [comment]
Added docblocks for @squeed

**Pull request merged**